tests/services/loading.service.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { LoadingService } from '../../src/app/core/services/loading.service';
+
+describe('LoadingService', () => {
+  describe('.show()', () => {
+    it('returns a value which indicate that the component is to be loading', () => {
+      const loader = new LoadingService();
+      loader.show().subscribe((isLoading) => expect(isLoading).toBe(true));
+    });
+  });
+
+  describe('.hide()', () => {
+    it('returns a value which indicates that the component is not to be loading', () => {
+      const loader = new LoadingService();
+      loader.hide().subscribe((isLoading) => expect(isLoading).toBe(false));
+    });
+  });
+
+  describe('.show() and .show()', () => {
```

Comment:
Is there a reason for testing the piped functions? The first two seem to be good enough

Reply:
These test cases are specifically to ensure that multiple successive calls have the same expected values. Given that the loading service would be used in cases where there could be multiple successive calls, i thought it was important to check that there were no race conditions

Reply:
i have changed the test cases now, it's a bit harder to test now esp because it's directly appending the spinner in the dom